crates/reporting/src/error/type.rs

```diff
@@ -3225,7 +3226,11 @@ fn should_show_diff(t1: &ErrorType, t2: &ErrorType) -> bool {
                 .any(|(t1, t2)| should_show_diff(t1, t2))
         }
         (Infinite, Infinite) | (Error, Error) => false,
-        (FlexVar(v1), FlexVar(v2)) | (RigidVar(v1), RigidVar(v2)) => v1 != v2,
+        (RigidVar(v1), RigidVar(v2)) => v1 != v2,
+        (FlexVar(_), _) | (_, FlexVar(_)) => {
+            // If either is flex, it will unify to the other type; no diff is needed.
+            false
```

Comment:
Does this sound right @ayazhafiz @folkertdev? In other words, if we have a type mismatch between two records, and we're trying to decide whether to show a field in the diff, is it valid to say "if the field on one side is a flex var, we don't need to show the diff because it will just end up being something like `foo : a` on one side of the diff and `foo : Str` on the other side, which isn't actually a diff" or am I missing an important case?

Reply:
Yes, this seems fine. The case you describe , of `a` vs `Str` for the same key, should never occur at this point though.